### PR TITLE
Add mercurial-app recipe (metapackage)

### PR DESCRIPTION
**Pull request by paugier**
This PR is mainly to receive feedback. My goal is to improve Mercurial packaging (mainly for Windows and macOS) with conda so that my students (and other people) can install Mercurial and important extensions without compilation with one (or few) conda command(s) or with the anaconda navigator. 

See:

- https://stackoverflow.com/questions/55657360/conda-and-conda-forge-to-install-commands-available-from-all-conda-environments
- https://github.com/conda/conda/issues/8556
- https://groups.google.com/a/continuum.io/forum/#!topic/conda/m34amfUW0Rc

Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages
- [x] Build number is 0
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/mercurial-app/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+app:
+  entry: hg
+  icon: logo.png
+  summary: Source control management tool
+  own_environment: True
+
+about:
+  home: https://github.com/conda-forge/mercurial-app-feedstock
+  license: GPL2
+  license_file: {{ RECIPE_DIR if RECIPE_DIR is defined else '' }}/LICENSE.txt
```
Comment (chrisburr) [inline recipes/mercurial-app/meta.yaml]: There is no need for the `RECIPE_DIR` part as conda automatically looks there if it isn't it the build directory
```suggestion
  license_file: LICENSE.txt
```

**Comment by paugier** [comment]
The mercurial package is already on conda-forge (https://github.com/conda-forge/mercurial-feedstock).

mercurial-app is a metapackage for a Mercurial "conda application" providing Mercurial (along with few important extensions).  It could allow conda users to install Mercurial as an application (i.e. not available only from one conda environment) on any platforms and without compilation.

Could such metapackages for conda applications be merged in conda-forge?

**Comment by paugier** [comment]
Hi @chrisburr, do I have to change anything else in this recipe?

**Comment by paugier** [comment]
@conda-forge/staged-recipes Can this PR be merged?

**Pull request merged**